core/src/driver/stablex_driver.rs

```diff
@@ -1,12 +1,27 @@
-use crate::metrics::StableXMetrics;
-use crate::models::{account_state::AccountState, order::Order, Solution};
-use crate::orderbook::StableXOrderBookReading;
-use crate::price_finding::PriceFinding;
-use crate::solution_submission::{SolutionSubmissionError, StableXSolutionSubmitting};
-use anyhow::{Error, Result};
+use crate::{
+    bigint_u256,
+    metrics::StableXMetrics,
+    models::{account_state::AccountState, order::Order, Solution},
+    orderbook::StableXOrderBookReading,
+    price_finding::PriceFinding,
+    solution_submission::{SolutionSubmissionError, StableXSolutionSubmitting},
+};
+use anyhow::{anyhow, Error, Result};
+use ethcontract::U256;
 use futures::future::{BoxFuture, FutureExt as _};
 use log::{info, warn};
-use std::time::{Duration, Instant};
+use num::{BigInt, Zero as _};
+use std::{
+    collections::HashMap,
+    time::{Duration, Instant},
+};
+
+// Used when we cannot calculate the amount of burned fees.
+const DEFAULT_GAS_PRICE_CAP: u64 = 200_000_000_000;
+// This is approximate. Depends on reversion of previous solution.
+const GAS_PER_TRADE: f64 = 120_000.0;
+// Will be removed in a future PR where we are going to fetch the current price.
```

Comment:
Just a suggestion, we already get price estimates, and we can re-use the WETH price estimate for this.